netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -2077,13 +2121,11 @@ net_ebpf_extension_sock_addr_redirect_connection_classify(
         compartment_id);
 
 Exit:
-    if (verdict == BPF_SOCK_ADDR_VERDICT_REJECT) {
-        // Create a blocked connection context and add it to list for the AUTH_CONNECT layer callout to enforce the
+    if (verdict != BPF_SOCK_ADDR_VERDICT_PROCEED) {
+        // Create a connection context and add it to list for the AUTH_CONNECT layer callout to enforce the
```

Comment:
what is the reason to change the behavior of storing only blocked connections?

Reply:
Programs are invoked in a callout in the connect-redirect layer, where each program determines the verdict of the connection. If the verdict isn't BPF_SOCK_ADDR_VERDICT_PROCEED, it is stored this list. Later, the callout in the auth-connect layer queries this list to determine the connection's verdict. If this list only stores blocked connections, and the callout at the auth-connect layer queries it but finds nothing, then it wouldn't be able to differentiate between BPF_SOCK_ADDR_VERDICT_PROCEED and BPF_SOCK_ADDR_VERDICT_PROCEED_HARD.

Reply:
Switching to allowed list adds some other complications. May need to discuss this a bit.